### PR TITLE
feat: Buy Success modal 

### LINKED ISSUE #8978: Success modal for mass buy
here we go, the first round of success modals 

after: 
- #8904 

image

## Design 
- updated on a success and signing modal page https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=6886-20058&mode=design&t=Tx1AQS2qpS8MaBtV-11
- it's a first iteration, let's see how this one performs

### Modal
- the top part is the same as on any other success modal 
- there is a change in how the "confirmed" label is shown 
image
- this is even more positive and its more clear that the tx is confirmed

### Body 
- then there is a list of items 
- The maximum default items preview is 6 items, if there is more, its hidden under the "show all" link
- if the show all link is clicked (it has hover effect -> black) it shows all the items :D 
   - the +X(items) is then hidden 

### Actions
- actions like share etc. similar to the claim modal on our drop page 
- The main actions are: 
   - view nfts - redirect to profile 
   - list nfts - our primary

### LINKED ISSUE #8980: Success modal claim drop main button
image
- text to black please

### LINKED ISSUE #8981: Single buy success modal
next one 

after: 
- #8904 

## Design
- updated here: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=6886-20058&mode=design&t=Tx1AQS2qpS8MaBtV-11
image

## TBD 
- reuse component that we use currently for drop claim
  - different copy - text to 16px 
  - confirmed tx different styling

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #8978
- [x] Closes #8981 
- [x] Closes #8980

## Needs QA check

- @kodadot/qa-guild please review

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI;

CleanShot 2024-02-02 at 14 20 58@2x

### CODE REVIEW

**Comment by exezbcz** [comment]
> * missing copy button( what does it copy tho?)

we agreed with hassnian that for mass buy it does not make much sense

**Comment by hassnian** [comment]
> round one(will try with more than 6 next time):
> 
>  mass.buying.mp4 
> * many notifications appearing
> * view nft does not close the modal
> * missing copy button( what does it copy tho?)

@prury  fixed

File: components/collection/drop/modal/shared/SuccessfulDrop.vue [hunk 1]
```diff
@@ -125,21 +23,23 @@ const props = defineProps<{
...
-    props.mintedNft.chain as Prefix,
-  ),
-)
+const txHash = computed(() => props.mintedNft.txHash || '')
+const share = computed(() => ({ text: sharingTxt, url: nftFullUrl.value }))
 
-const collectionUrl = computed(
-  () => `/${props.mintedNft.chain}/collection/${props.mintedNft.collection}`,
-)
+const actionButtons = computed(() => ({
```
Comment (preschian) [inline components/collection/drop/modal/shared/SuccessfulDrop.vue:32]: Could you check if it still works without `computed`?

File: components/common/shoppingCart/utils.ts [hunk 2]
```diff
@@ -84,3 +84,19 @@ export const nftToListingCartItem = (
...
+): ListCartItem => {
+  return {
+    id: item.id,
+    name: item.name,
+    price: item.price ?? '0',
+    urlPrefix: item.urlPrefix,
+    collection: { ...item.collection, floor },
+    metadata: item.metadata,
+    meta: item.meta,
+    listPrice: null,
```
Comment (preschian) [inline components/common/shoppingCart/utils.ts:100]: Type 'null' is not assignable to type 'number'.

File: libs/ui/src/components/NeoButton/NeoButton.scss [hunk 3]
```diff
@@ -30,7 +30,15 @@
     }
 
     &--k-accent {
-      @apply bg-k-primary text-black;
+      @apply bg-k-primary ;
+    
+      &:not(:hover):not(:disabled) {
+        @apply text-black #{!important};
+      }
```
Comment (preschian) [inline libs/ui/src/components/NeoButton/NeoButton.scss:37]:
```suggestion
        @apply !text-black;
```

File: libs/ui/src/components/NeoButton/NeoButton.scss [hunk 4]
```diff
@@ -30,7 +30,15 @@
...
+
+      &:enabled {
+        @apply border-border-color #{!important};
+      }
```
Comment (preschian) [inline libs/ui/src/components/NeoButton/NeoButton.scss:41]:
```suggestion
        @apply !border-border-color;
```

**Review by preschian — APPROVED** [review]
otherwise lgtm

File: libs/ui/src/components/NeoButton/NeoButton.scss (hunk 3 above)
Comment (hassnian) [reply libs/ui/src/components/NeoButton/NeoButton.scss:37]: getting this 

``````
[plugin:vite:css] [sass] expected ";".
   ╷
36 │         @apply !text-black;
   │                ^
   ╵
  libs/ui/src/components/NeoButton/NeoButton.scss 36:16  @import
  libs/ui/src/components/NeoButton/NeoButton.vue 2:9     root stylesheet

`````

![CleanShot 2024-02-05 at 10 16 44@2x](https://github.com/kodadot/nft-gallery/assets/44554284/0f008a0c-ed6d-43f5-a73d-0fdd0bb65b02)

Comment (preschian) [reply libs/ui/src/components/NeoButton/NeoButton.scss:37]: ouch, lgtm then 👍🏻

File: components/common/shoppingCart/utils.ts (hunk 2 above)
Comment (hassnian) [reply components/common/shoppingCart/utils.ts]: fixed

File: components/collection/drop/modal/shared/SuccessfulDrop.vue (hunk 1 above)
Comment (hassnian) [reply components/collection/drop/modal/shared/SuccessfulDrop.vue:32]: checked to be sure , it doens't , `cantList` is not updated if `actionButtons ` it's just an object. 

CleanShot 2024-02-05 at 09 56 36@2x

File: components/common/shoppingCart/utils.ts (hunk 2 above)
Comment (hassnian) [reply components/common/shoppingCart/utils.ts]: fixed

**Comment by prury** [comment]
double slash when pasting the address copied from success buy and when sharing to telegram(singlebuy): 
image

first try showed me this error: 

https://github.com/kodadot/nft-gallery/assets/36627808/8a16861e-2342-42f9-a9f3-ad3c65db379b

second try(worked): 

https://github.com/kodadot/nft-gallery/assets/36627808/3fead7de-e027-42e7-a283-69aef4192bca

**Comment by hassnian** [comment]
> double slash when pasting the address copied from success buy and when sharing to telegram(singlebuy): image
> 

fixed

> first try showed me this error:
> 
> https://assethub-kusama.subscan.io/tx/0x445d9cb53a932e5e97aace80f0f91399531dc80c5b0835d12830a6560b9c8f0c
> 

should be unrelated to this pr . please create an issue if it persists, I'll have a look 

thanks

File: components/buy/SuccessfulBuyModal.vue [hunk 5]
```diff
@@ -0,0 +1,176 @@
+<template>
```
Comment (roiLeo) [inline components/buy/SuccessfulBuyModal.vue:1]: IMO this need to be wrapped in a `div`

File: components/buy/SuccessfulBuyModal.vue [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        <p class="is-size-6 capitalize font-bold text-center">
+          {{ $t('buyModal.amountPurchaseSuccessfully', [items.length]) }}
+        </p>
+      </div>
+    </template>
+  </SuccessfulModal>
+
+  <ListingCartModal />
+</template>
+<script setup lang="ts">
```
Comment (roiLeo) [inline components/buy/SuccessfulBuyModal.vue:63]:
```suggestion
</template>

<script setup lang="ts">
```

**Comment by prury** [comment]
maybe new bug got introduced? i don't see this NaN on canary when listing items

image

**Comment by hassnian** [comment]
> maybe new bug got introduced? i don't see this NaN on canary when listing items
> 
> 
> 
> image
> 
> 

hmm will check tomorrow morning

File: components/common/listingCart/shared/ListingCartPriceInput.vue [hunk 7]
```diff
@@ -31,8 +31,17 @@ const props = defineProps<{
...
+const emit = defineEmits(['confirm', 'update:modelValue'])
+
+const model = computed({
+  get() {
+    return props.modelValue
+  },
+  set(value) {
+    emit('update:modelValue', value === '' ? undefined : value)
+  },
+})
```
Comment (roiLeo) [inline components/common/listingCart/shared/ListingCartPriceInput.vue:43]:
```suggestion
  const model = computed({
    get: () => props.modelValue,
    set: (value) => { 
      emit('update:modelValue', value === '' ? undefined : value)
    },
  })
```

File: components/common/successfulModal/SingleItemMedia.vue [hunk 8]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <BaseMediaItem class="border border-k-shade" :src="src" preview is-detail />
```
Comment (roiLeo) [inline components/common/successfulModal/SingleItemMedia.vue:2]: wrap it in `div`

File: components/common/successfulModal/SingleItemMedia.vue [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+  <div class="mt-5 border-b-k-shade">
+    <p class="is-size-6 capitalize font-bold text-center">
+      {{ header }}
+    </p>
+    <p class="capitalize text-xs text-center mt-2">
+      {{ $t('drops.artBy', [nftName]) }}
+      <a
+        v-safe-href="collectionUrl"
```
Comment (roiLeo) [inline components/common/successfulModal/SingleItemMedia.vue:11]: please use `NuxtLink`, no need for computed property

File: components/common/successfulModal/SuccessfulModalBody.vue [hunk 10]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <TransactionSection :tx-hash="txHash" class="mb-5" />
```
Comment (roiLeo) [inline components/common/successfulModal/SuccessfulModalBody.vue:2]: wrap all in a single `div`

File: components/common/successfulModal/TransactionSection.vue [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        pack="fass"
+        size="small" />
+
+      <NeoIcon
+        v-clipboard:copy="txUrl"
+        icon="copy"
+        pack="fass"
+        class="text-k-grey cursor-pointer"
+        data-testid="tx-clipboard"
+        @click="toast($t('general.copyToClipboard'))" />
```
Comment (roiLeo) [inline components/common/successfulModal/TransactionSection.vue:33]: please no click event on Icon, use button instead

File: components/buy/SuccessfulBuyModal.vue (hunk 5 above)
Comment (hassnian) [reply components/buy/SuccessfulBuyModal.vue:1]: fixed

File: components/common/successfulModal/TransactionSection.vue (hunk 11 above)
Comment (hassnian) [reply components/common/successfulModal/TransactionSection.vue]: fixed

File: components/common/successfulModal/SuccessfulModalBody.vue (hunk 10 above)
Comment (hassnian) [reply components/common/successfulModal/SuccessfulModalBody.vue]: fixed

File: components/common/successfulModal/SingleItemMedia.vue (hunk 9 above)
Comment (hassnian) [reply components/common/successfulModal/SingleItemMedia.vue]: fixed

File: components/common/successfulModal/SingleItemMedia.vue (hunk 8 above)
Comment (hassnian) [reply components/common/successfulModal/SingleItemMedia.vue]: fixed

File: components/common/listingCart/shared/ListingCartPriceInput.vue (hunk 7 above)
Comment (hassnian) [reply components/common/listingCart/shared/ListingCartPriceInput.vue]: fixed

**Comment by hassnian** [comment]
> maybe new bug got introduced? i don't see this NaN on canary when listing items

@prury fixed 

CleanShot 2024-02-07 at 16 36 55@2x

**Comment by prury** [comment]
buy went fine, only notifications showing up again: 
https://github.com/kodadot/nft-gallery/assets/36627808/8650f172-d2c7-42e4-8620-d8ab5a9a1e5d

@hassnian new issue for notifications?

**Review by roiLeo — APPROVED** [review]

**Comment by hassnian** [comment]
> buy went fine, only notifications showing up again: 
> 
> 
> 
> https://github.com/kodadot/nft-gallery/assets/36627808/8650f172-d2c7-42e4-8620-d8ab5a9a1e5d
> 
> 
> 
> @hassnian new issue for notifications? 
> 

Thought it was fixed ahh 

we already have an issue #9272 let's continue there

**Comment by prury** [comment]
> > buy went fine, only notifications showing up again:
> > 
> >   
> >     
> >     
> > 
> >     massbuy.modal.3.mp4
> >     
> >   
> > 
> >   
> > 
> >   
> > 
> > 
> > @hassnian new issue for notifications?
> 
> Thought it was fixed ahh
> 
> we already have an issue #9272 let's continue there

okok

**Comment by yangwao** [comment]
Thanks!
pay 90 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $90 @ 6.78 USD/DOT ~ 13.274 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0x65a316876115565c8cfba2318173c7a1145d34221bbd7904fbf81bc3e6595a89](https://polkadot.subscan.io/extrinsic/0x65a316876115565c8cfba2318173c7a1145d34221bbd7904fbf81bc3e6595a89)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**